llvm/test/Transforms/LoadStoreVectorizer/NVPTX/256-bit.ll

```diff
@@ -0,0 +1,728 @@
+  %ptr1 = getelementptr double, ptr addrspace(1) %ptr, i64 1
+  %ptr2 = getelementptr double, ptr addrspace(1) %ptr, i64 2
+  %ptr3 = getelementptr double, ptr addrspace(1) %ptr, i64 3
+
+  %l0 = load double, ptr addrspace(1) %ptr0, align 32
+  %l1 = load double, ptr addrspace(1) %ptr1, align 1
+  %l2 = load double, ptr addrspace(1) %ptr2, align 16
+  %l3 = load double, ptr addrspace(1) %ptr3, align 1
+
+  store double %l3, ptr addrspace(1) %ptr0, align 32
+  store double %l2, ptr addrspace(1) %ptr1, align 1
+  store double %l1, ptr addrspace(1) %ptr2, align 16
+  store double %l0, ptr addrspace(1) %ptr3, align 1
+
+  ret void
+}
+
+define void @int32x8_non_global(ptr %ptr) {
+; CHECK-LABEL: define void @int32x8_non_global(
+; CHECK-SAME: ptr [[PTR:%.*]]) #[[ATTR0:[0-9]+]] {
+; CHECK-NEXT:    [[PTR0:%.*]] = getelementptr i32, ptr [[PTR]], i64 0
+; CHECK-NEXT:    [[PTR4:%.*]] = getelementptr i32, ptr [[PTR]], i64 4
+; CHECK-NEXT:    [[TMP1:%.*]] = load <4 x i32>, ptr [[PTR0]], align 32
+; CHECK-NEXT:    [[L01:%.*]] = extractelement <4 x i32> [[TMP1]], i32 0
+; CHECK-NEXT:    [[L12:%.*]] = extractelement <4 x i32> [[TMP1]], i32 1
+; CHECK-NEXT:    [[L23:%.*]] = extractelement <4 x i32> [[TMP1]], i32 2
+; CHECK-NEXT:    [[L34:%.*]] = extractelement <4 x i32> [[TMP1]], i32 3
+; CHECK-NEXT:    [[TMP2:%.*]] = load <4 x i32>, ptr [[PTR4]], align 16
+; CHECK-NEXT:    [[L45:%.*]] = extractelement <4 x i32> [[TMP2]], i32 0
+; CHECK-NEXT:    [[L56:%.*]] = extractelement <4 x i32> [[TMP2]], i32 1
+; CHECK-NEXT:    [[L67:%.*]] = extractelement <4 x i32> [[TMP2]], i32 2
+; CHECK-NEXT:    [[L78:%.*]] = extractelement <4 x i32> [[TMP2]], i32 3
+; CHECK-NEXT:    [[TMP3:%.*]] = insertelement <4 x i32> poison, i32 [[L78]], i32 0
+; CHECK-NEXT:    [[TMP4:%.*]] = insertelement <4 x i32> [[TMP3]], i32 [[L67]], i32 1
+; CHECK-NEXT:    [[TMP5:%.*]] = insertelement <4 x i32> [[TMP4]], i32 [[L56]], i32 2
+; CHECK-NEXT:    [[TMP6:%.*]] = insertelement <4 x i32> [[TMP5]], i32 [[L45]], i32 3
+; CHECK-NEXT:    store <4 x i32> [[TMP6]], ptr [[PTR0]], align 32
+; CHECK-NEXT:    [[TMP7:%.*]] = insertelement <4 x i32> poison, i32 [[L34]], i32 0
+; CHECK-NEXT:    [[TMP8:%.*]] = insertelement <4 x i32> [[TMP7]], i32 [[L23]], i32 1
+; CHECK-NEXT:    [[TMP9:%.*]] = insertelement <4 x i32> [[TMP8]], i32 [[L12]], i32 2
+; CHECK-NEXT:    [[TMP10:%.*]] = insertelement <4 x i32> [[TMP9]], i32 [[L01]], i32 3
+; CHECK-NEXT:    store <4 x i32> [[TMP10]], ptr [[PTR4]], align 16
+; CHECK-NEXT:    ret void
+
+  %ptr0 = getelementptr i32, ptr %ptr, i64 0
+  %ptr1 = getelementptr i32, ptr %ptr, i64 1
+  %ptr2 = getelementptr i32, ptr %ptr, i64 2
+  %ptr3 = getelementptr i32, ptr %ptr, i64 3
+  %ptr4 = getelementptr i32, ptr %ptr, i64 4
+  %ptr5 = getelementptr i32, ptr %ptr, i64 5
+  %ptr6 = getelementptr i32, ptr %ptr, i64 6
+  %ptr7 = getelementptr i32, ptr %ptr, i64 7
+
+  %l0 = load i32, ptr %ptr0, align 32
+  %l1 = load i32, ptr %ptr1, align 1
```

Comment:
Hmm.  `align 1` looks rather odd here. We know that `ptr0` is aligned by 32. And we know that `ptr1` is offset by 4, so correct alignment must be 4. And so on for the other elements.
Correct alignment info should be: 32,4,8,4,16,4,8,4

Looks like vectorize was able to figure it out, and vectorized the loads/stores, but it would be great if we'd provide sensible inputs.